src/components/backup/RestoreCloudStep.tsx

```diff
@@ -199,17 +197,19 @@ export default function RestoreCloudStep() {
             }
           });
 
-          await dispatch(setAllWalletsWithIdsAsBackedUp(oldCloudIds, walletBackupTypes.cloud, filename));
-          await dispatch(setAllWalletsWithIdsAsBackedUp(oldManualIds, walletBackupTypes.manual, filename));
+          setAllWalletsWithIdsAsBackedUp(oldCloudIds, walletBackupTypes.cloud, filename);
+          setAllWalletsWithIdsAsBackedUp(oldManualIds, walletBackupTypes.manual, filename);
 
           const walletKeys = Object.keys(newWalletsState || {});
-          // @ts-expect-error TypeScript doesn't play nicely with Redux types here
           const firstWallet = walletKeys.length > 0 ? (newWalletsState || {})[walletKeys[0]] : undefined;
           const firstAddress = firstWallet ? (firstWallet.addresses || [])[0].address : undefined;
-          const p1 = dispatch(walletsSetSelected(firstWallet));
-          const p2 = dispatch(addressSetSelected(firstAddress));
-          await Promise.all([p1, p2]);
-          await initializeWallet(null, null, null, false, false, null, true, null);
+
+          if (firstWallet && firstAddress) {
+            const { setSelectedAddress, setSelectedWallet } = useWalletsStore.getState();
+            setSelectedWallet(firstWallet);
```

Comment:
same here, can we consolidate into: `setSelectedWallet(firstWallet, firstAddress)`